js/friday.js

```diff
@@ -1,34 +1,38 @@
 (function IsItFridayYet() {
   const date = new Date();
-  const theMonth = date.getMonth();
-  const theDay = date.getDay();
-  const element = document.getElementById("friday");
-  const spans = document.getElementsByClassName("fallingElement");
+  const thisMonth = date.getMonth();
+  const thisDay = date.getDay();
+  const theBody = document.body;
+  const textElement = document.getElementById("friday");
   let season = "";
 
-  if (theDay === 5) {
-    element.innerHTML = "YES";
+  // return a programatic yes on Fridays to print on homepage.
+  function isFriday(n) {
+    return (n == 5 ? "YES" : "NO");
+  }
 
-    switch (true) {
-      case theMonth >= 3 && theMonth < 6:
-        season = "spring";
-        break;
-      case theMonth >= 6 && theMonth < 9:
-        season = "summer";
-        break;
-      case theMonth >= 9 && theMonth < 12:
-        season = "autumn";
-        document.body.style.backgroundColor = "fff7c8";
-        break;
-      default:
-        season = "winter";
-        document.body.style.backgroundColor = "dcf0f4";
-        break;
+  // return a season string to switch up body class later.
+  function thisSeason(range) {
+    if(range >= 2 && range < 5 ) {
+      return "spring";
+    } else if(range >= 5 && range < 8 ) {
+      return "summer";
+    } else if(range >= 8 && range < 11) {
+      return "fall";
+    } else if(range >= 11 || range <= 2) {
+      return "winter";
+    } else {
+      return "";
     }
-    Array.from(spans).forEach(function (element) {
-      element.classList.add(season);
-    });
-  } else {
-    element.innerHTML = "NO";
   }
+
+  // restyle the site based on the season because we can dammit.
+  function seasonStyles() {
```

Comment:
Now our functions get wild.

In seasonStyles we first set the let variable of season to be equal to a function that will return us the string value of the season based on the numbered month we pass as an argument.

Once that is set, we then add the class of that season to the body element.

If you really really look at these functions, you can argue that its a tad bit verbose, however, what if there is no internet connection? what if one function fails. Does the site stop running? Ideally not. That's the point. That's how we tackle and learn to master our JavaScript.